callstack/react-native-paper
# feat: adjust typography component, fix theme types

### Summary

This PR introduces a new way of approaching typography - a `<Text>` component with multiple `variants` that are Material 3 compliant

### Usage

`<Text variant="heading-large" />`

### Test plan

1. Open the example app
2. Navigate to Typography
3. You will see old examples at the top (still supported)
4. You will see new examples at the bottom (`<Text variant="...">` usage)

![image](https://user-images.githubusercontent.com/24693481/152163340-60d4cd24-76b5-4ef3-b3e2-23d2fcb35dcd.png)
